packages/@aws-cdk/cloudformation-include/lib/cfn-include.ts

```diff
@@ -231,6 +289,74 @@ export class CfnInclude extends core.CfnElement {
     this.conditions[conditionName] = cfnCondition;
   }
 
+  private createNestedStack(nestedStackId: string, props?: CfnIncludeProps): void {
+    const resources = this.template.Resources;
+
+    if (!(nestedStackId in resources)) {
+      throw new Error(`Nested Stack with logical ID '${nestedStackId}' was not found in the template`);
+    }
+
+    if (!(resources[nestedStackId].Type === 'AWS::CloudFormation::Stack' && props?.nestedStacks)) {
+      throw new Error(`Nested Stack with logical ID '${nestedStackId}' is not an AWS::CloudFormation::Stack resource`);
+    }
+
+    const self = this;
+    const cfnParser = new cfn_parse.CfnParser({
+      finder: {
+        findCondition(conditionName: string): core.CfnCondition | undefined {
+          return self.conditions[conditionName];
+        },
+
+        findResource(lId: string): core.CfnResource | undefined {
+          if (!(lId in (self.template.Resources || {}))) {
+            return undefined;
+          }
+          return self.getOrCreateResource(lId);
+        },
+
+        findRefTarget(elementName: string): core.CfnElement | undefined {
+          if (elementName in self.parameters) {
+            return self.parameters[elementName];
+          }
+
+          return this.findResource(elementName);
+        },
+      },
+    });
+
+    const nestedStackAttributes = resources[nestedStackId];
+    const nestedStackProps = cfnParser.parseValue(nestedStackAttributes.Properties);
+
+    const nestedStackScope = new core.NestedStack(this, nestedStackId, {
+      parameters: nestedStackProps.Parameters,
+      notificationArns: nestedStackProps.NotificationArns,
+      timeout: nestedStackProps.Timeout,
+    });
+
+    const cfnOptions = nestedStackScope.nestedStackResource!.cfnOptions;
+    cfnOptions.metadata = cfnParser.parseValue(nestedStackAttributes.Metadata);
+    cfnOptions.deletionPolicy = cfnParser.parseDeletionPolicy(nestedStackAttributes.DeletionPolicy);
+    cfnOptions.updateReplacePolicy = cfnParser.parseDeletionPolicy(nestedStackAttributes.UpdateReplacePolicy);
+    nestedStackAttributes.dependsOn = resources.DependsOn ?? [];
```

Comment:
I'm sorry, what is this...? We need proper handling of the `DependsOn` attribute.

Take a look at something like the `CfnBucket.fromCloudFormation()` method to see how this works.

We also need unit tests confirming this works.